app/manage/pods/device-details/share-modal-directive/index.js

```diff
@@ -11,10 +11,12 @@ module.exports = [
       controller: [
         '$scope',
         '$timeout',
-        function($scope, $timeout) {
+        '$location',
+        function($scope, $timeout, $location) {
+
           // template variables -------------------
           $scope.copySuccess = false;
-          $scope.url = window.location.hostname + window.location.pathname;
+          $scope.url = $location.absUrl().split('?')[0];
```

Comment:
hrmm that might actually be a problem.  What I'm trying to get rid of is the ?demo=1  but keep the hash. 

so going to `http://localhost:5000/manage/#/device/9544d25e-0565-4f27-9b66-870b0c02d471?demo=1` will link to ``http://localhost:5000/manage/#/device/9544d25e-0565-4f27-9b66-870b0c02d471`

Maybe I should specifically match on demo=1